include/meta/meta.hpp

```diff
@@ -961,6 +968,49 @@ namespace meta
     template <typename List, typename Predicate>
     using filter = meta::foldl<List, meta::list<>, detail::filter_<Predicate>>;
 
+    ////////////////////////////////////////////////////////////////////////////
+    // static_const
+    ///\cond
+    namespace detail
+    {
+        template <typename T> struct static_const
+        {
+            static constexpr T value{};
+        };
+
+        // Avoid potential ODR violations with global objects:
+        template <typename T> constexpr T static_const<T>::value;
+    } // namespace detail
+
+    ///\endcond
+
+    ////////////////////////////////////////////////////////////////////////////
+    // for_each
+    /// \cond
+    namespace detail
+    {
+        struct for_each_fn
+        {
+            template <class UnaryFunction, class... Args>
+            constexpr auto operator()(meta::list<Args...>,
+                                      UnaryFunction &&f) const
+              -> decltype(std::forward<UnaryFunction>(f))
+            {
+                return (void)std::initializer_list<int>{(f(Args{}), 0)...},
+                       std::forward<UnaryFunction>(f);
```

Comment:
Two more things, sorry. The STL takes and (in the case of `std::for_each`, returns) function objects by value. We should too. You can nuke the `std::forward` business. Also, the expression can be made a little more bulletproof like:  `(f(Args{}), void(), 0)...`. The `void()` guards against perverse comma operator overloads.